src/astro/databases/google/bigquery.py

```diff
@@ -236,3 +257,177 @@ def gs_to_bigquery(
         self.hook.insert_job(
             configuration=job_config,
         )
+
+    def s3_to_bigquery(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        Transfer data from S3 to Bigquery via invoking datatransfer job
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+        """
+        native_support_kwargs = native_support_kwargs or {}
+
+        project_id = self.get_project_id(target_table)
+
+        if if_exists == "replace":
+            # We need to create an empty table as datatransfer job can only append to an existing table,
+            # so we need to create a table.
+            self.create_empty_table(source_file, target_table)
+
+        transfer = S3ToBigqueryDataTransfer(
+            target_table=target_table,
+            source_file=source_file,
+            project_id=project_id,
+            native_support_kwargs=native_support_kwargs,
+            **kwargs,
+        )
+        transfer.run()
+
+
+class S3ToBigqueryDataTransfer:
+    """
+    Create and run Datatransfer job from S3 to Bigquery
+    """
+
+    def __init__(
+        self,
+        target_table: Table,
+        source_file: File,
+        project_id: str,
+        poll_duration: int = 1,
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param project_id: Bigquery project id
+        :param poll_duration: sleep duration between two consecutive job status checks. Unit - seconds. Default 1 sec.
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+        """
+        self.client = BiqQueryDataTransferServiceHook(gcp_conn_id=target_table.conn_id)
+        self.target_table = target_table
+        self.source_file = source_file
+
+        conn = BaseHook.get_connection(source_file.conn_id)
```

Comment:
I exposed the hook in the FileLocation class - it may help here:
https://github.com/astronomer/astro-sdk/pull/523/files#diff-aaa4e821af24c22f598e2399b6d9afc54358185f4810740bc387b966f87da14eR18

Reply:
Duplicate.